### PR TITLE
Add a SkipRetrySignal option to our archival client

### PR DESCRIPTION
**Pull request by MichaelSnowden**
**What changed?**
I added an option to our archival client to bypass our retry logic which currently sends a signal to the archival workflow to retry for each target we failed to archive inline. Bypassing this logic means we don't retry here. 

**Why?**
This option is needed for durable archival, so that we can define our own retry logic. In durable archival, retries will be handled by the archival queue itself, not us signaling an resignaling the archival workflow.

**How did you test it?**
I verified this change by adding tests that run the existing test suite with this flag on, verifying that we skip sending the archival signal in that case, and we return a multi err containing the errors of the exact targets that failed.

**Potential risks**
I don't think anything because this flag defaults to false, and the actual changes to the default branch are just adding the error to a multierr. If something panics there, we would have a problem that would prevent archival from working.

**Is hotfix candidate?**
No

### CODE REVIEW

File: service/worker/archiver/client.go [hunk 1]
```diff
@@ -56,6 +57,11 @@ type (
 		ArchiveRequest       *ArchiveRequest
 		CallerService        string
 		AttemptArchiveInline bool
+		// SkipRetrySignal will skip sending a signal to the archival workflow to retry archival if the inline attempt
+		// fails. Note that this does nothing if AttemptArchiveInline is false. We will set this to true for durable
+		// archival because the archival queue handles retries, and then we can  remove this condition when the old
+		// behavior is deleted.
+		SkipRetrySignal bool
```
Comment (yux0) [inline service/worker/archiver/client.go:64]: Can the send signal logic remove after durable archival? Is this flag needed?

Comment (MichaelSnowden) [reply service/worker/archiver/client.go:64]: This flag is needed, but we can remove it after durable archival is fully rolled out.

Comment (yycptt) [reply service/worker/archiver/client.go:64]: Once we have durable archival, the code should be moved to history service. and archival client probably should not exist at all.

so I guess, we can either 
- add the flag, use the client for now in archival queue, then later refactor. 
- Or just copy/paste the inline archival code and use it in archival task executor. Then later we can simply delete the entire archival client. 
Either way we'll need to do some refactoring. so I don't have strong opinion on which one we should pick.

Comment (yux0) [reply service/worker/archiver/client.go:64]: Yeah. I would vote for the later. With refactoring, this seems a throwing work.

**Pull request closed without merging**